users/tests/test_user_login.py

```diff
@@ -22,3 +33,29 @@ def test_user_client_should_not_log_in_when_email_is_none(self):
 
     def test_user_client_should_not_log_in_when_password_is_none(self):
         self.assertFalse(self.client.login(email="testuser@codepoets.it", password=None))
+
+    def test_user_should_redirect_to_change_password_form_if_need_change_password_is_true(self):
+        self._set_need_change_password(need_change_password=True)
+
+        response = self.client.post(reverse("login"), data=self.data)
+        self.assertRedirects(response, reverse("login_password_change"), status_code=302, target_status_code=200)
+
+    def test_user_should_redirect_to_home_if_need_change_password_is_false(self):
+        self._set_need_change_password(need_change_password=False)
+
+        response = self.client.post(reverse("login"), data=self.data)
+        self.assertRedirects(response, reverse("home"), status_code=302, target_status_code=200)
+
+    def test_user_with_correct_email_and_password_should_log_in_on_post(self):
+        response = self.client.post(reverse("login"), data=self.data)
+        self.assertEqual(response.status_code, 302)
+        self.assertTrue(self.user.is_authenticated)
+
+    def test_login_form_use_proper_template(self):
+        response = self.client.get(reverse("login"))
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, "registration/login.html")
```

Comment:
I can't see purpose of this test. Can you explain that?

Reply:
Probably I misunderstood that we need test for template used by the login form.

Reply:
Deleted.